src/main/java/core/di/factory/BeanScanner.java

```diff
@@ -4,53 +4,42 @@
 import core.annotation.Repository;
 import core.annotation.Service;
 import core.annotation.web.Controller;
-import core.annotation.web.RequestMapping;
-import core.mvc.tobe.HandlerExecution;
-import core.mvc.tobe.HandlerKey;
-import core.mvc.tobe.support.*;
 import org.reflections.Reflections;
 import org.reflections.scanners.MethodAnnotationsScanner;
 import org.reflections.scanners.SubTypesScanner;
 import org.reflections.scanners.TypeAnnotationsScanner;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.core.LocalVariableTableParameterNameDiscoverer;
-import org.springframework.core.ParameterNameDiscoverer;
 
 import java.lang.annotation.Annotation;
-import java.lang.reflect.Method;
-import java.util.*;
-
-import static java.util.Arrays.asList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
 
 public class BeanScanner {
-
     private static final Logger logger = LoggerFactory.getLogger(BeanScanner.class);
 
-    private static final List<ArgumentResolver> argumentResolvers = asList(
-            new HttpRequestArgumentResolver(),
-            new HttpResponseArgumentResolver(),
-            new RequestParamArgumentResolver(),
-            new PathVariableArgumentResolver(),
-            new ModelArgumentResolver()
-    );
-
-    private static final ParameterNameDiscoverer nameDiscoverer = new LocalVariableTableParameterNameDiscoverer();
+    private Set<Class<?>> preInstanticateBeans;
+    private BeanFactory beanFactory;
 
-    public Map<HandlerKey, HandlerExecution> scan(Object... basePackage) {
+    public BeanScanner(Object... basePackage) {
         Reflections reflections = new Reflections(basePackage, new TypeAnnotationsScanner(), new SubTypesScanner(), new MethodAnnotationsScanner());
-        Map<HandlerKey, HandlerExecution> handlers = new HashMap<>();
+        this.preInstanticateBeans = getTypesAnnotatedWith(reflections, Controller.class, Service.class, Repository.class);
```

Comment:
빈 스캔 대상 클래스에 붙이는 애너테이션을 하나 정의하여
Controller, Service, Repository 에서 사용하게 한다면 어떨까요?
현재 구조라면 빈 관리 대상 애너테이션이 늘어날 때 마다 이 부분의 코드 수정이 불가피하겠네요ㅠ

참고로 아래 코드는 스프링의 `@Service` 애너테이션 입니다
```java
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Component
public @interface Service {

	/**
	 * The value may indicate a suggestion for a logical component name,
	 * to be turned into a Spring bean in case of an autodetected component.
	 * @return the suggested component name, if any (or empty String otherwise)
	 */
	@AliasFor(annotation = Component.class)
	String value() default "";

}

```